test/03_USM.test.js

```diff
@@ -0,0 +1,310 @@
+            // Deploy contracts
+            oracle = await TestOracle.new(price, shift, {from: deployer});
+            usm = await USM.new(oracle.address, {from: deployer});
+            fum = await FUM.at(await usm.fum());
+
+            // Setup variables
+            // etherPrice = parseInt(await oracle.latestPrice());
+            // etherPriceDecimalShift = parseInt(await oracle.decimalShift());
+            // ethDeposit = 100;
+            // mintAmount = "24975000000000000000000";
+
+            // Mint
+            // await usm.mint({from: user, value: toWei(ethDeposit)});
+        });
+
+        describe("deployment", () => {
+            it("sets latest fum price", async () => {
+                let latestFumPrice = (await usm.latestFumPrice()).toString();
+                let MIN_ETH_AMOUNT = (await usm.MIN_ETH_AMOUNT()).toString();
+                latestFumPrice.should.equal(MIN_ETH_AMOUNT);
+            });    
+        });
+
+        describe("minting and burning", () => {
+            it("allows minting USM", async () => {
+                const oneEth = WAD;
+                const fumPrice = (await usm.latestFumPrice()).toString();
+
+                await usm.mint({ from: user, value: oneEth });
+                const usmBalance = (await usm.balanceOf(user)).toString();
+                usmBalance.should.equal(oneEth.mul(priceWAD).div(WAD).toString());
+
+                const newFumPrice = (await usm.latestFumPrice()).toString();
+                newFumPrice.should.equal(fumPrice); // Minting doesn't change the fum price if buffer is 0
+            })
+
+            it("doesn't allow minting with less than MIN_ETH_AMOUNT", async () => {
+                const MIN_ETH_AMOUNT = await usm.MIN_ETH_AMOUNT();
+                // TODO: assert MIN_ETH_AMOUNT > 0
+                await expectRevert(
+                    usm.mint({ from: user, value: MIN_ETH_AMOUNT.sub(new BN('1')) }),
+                    "Must deposit more than 0.001 ETH"
+                );
+            })
+
+            describe("with existing USM supply", () => {
+                beforeEach(async () => {
+                    const oneEth = WAD;
+                    await usm.mint({ from: user, value: oneEth });
+                });
+
+                it("allows burning USM", async () => {
+                    const usmBalance = (await usm.balanceOf(user)).toString();
+                    const fumPrice = (await usm.latestFumPrice()).toString();
+    
+                    await usm.burn(usmBalance, { from: user });
+                    const newUsmBalance = (await usm.balanceOf(user)).toString();
+                    newUsmBalance.should.equal('0');
+                    // TODO: Test the eth balance just went up. Try setting gas price to zero for an exact amount
+    
+                    const newFumPrice = (await usm.latestFumPrice()).toString();
+                    newFumPrice.should.equal(fumPrice); // Burning doesn't change the fum price if buffer is 0
```

Comment:
USM `mints` and `burns` should never move the FUM price at all, except two cases:
1. If the system is collecting fees, they go to FUM holders, so will (slightly) increase the FUM price.
2. As long as debt ratio ≤ 100%, `burns` decrease debt ratio.  (If debt ratio > 100%, `burns` should be disallowed.)  So it's possible for a `burn` to pull debt ratio < max, causing `minFumBuyPrice` to be cleared and changing the effective FUM buy price back to its theoretical value (buffer / FUM qty) - which might be lower than the previous `minFumBuyPrice` (or might not).

Reply:
(Similarly, fees and crossings of `MAX_DEBT_RATIO` aside, `funds` and `defunds` shouldn't move FUM price either!  It mainly just moves when ETH price does.)